services/appflowy-worker/src/application.rs

```diff
@@ -0,0 +1,174 @@
+use aws_sdk_s3::config::{Credentials, Region, SharedCredentialsProvider};
+
+use crate::import_worker::email_notifier::EmailNotifier;
+use crate::s3_client::S3ClientImpl;
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::routing::get;
+use axum::{Json, Router};
+use secrecy::ExposeSecret;
+use serde_json::json;
+use std::sync::{Arc, Once};
+use std::time::Duration;
+use tokio::net::TcpListener;
+use tokio::task::LocalSet;
+use tracing::info;
+use tracing::subscriber::set_global_default;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::EnvFilter;
+
+pub async fn run_server(
+  listener: TcpListener,
+  config: Config,
+) -> Result<(), Box<dyn std::error::Error>> {
+  dotenvy::dotenv().ok();
+  init_subscriber(&config.app_env);
+  info!("config loaded: {:?}", &config);
+
+  // Start the server
+  info!("Starting server at: {:?}", listener.local_addr());
+  create_app(listener, config).await.unwrap();
+  Ok(())
+}
+
+pub fn init_subscriber(app_env: &Environment) {
+  static START: Once = Once::new();
+  START.call_once(|| {
+    let level = std::env::var("RUST_LOG").unwrap_or("info".to_string());
+    let mut filters = vec![];
+    filters.push(format!("appflowy_worker={}", level));
+    let env_filter = EnvFilter::new(filters.join(","));
+
+    let builder = tracing_subscriber::fmt()
+      .with_target(true)
+      .with_max_level(tracing::Level::TRACE)
+      .with_thread_ids(false)
+      .with_file(false);
+
+    match app_env {
+      Environment::Local => {
+        let subscriber = builder
+          .with_ansi(true)
+          .with_target(false)
+          .with_file(false)
+          .pretty()
+          .finish()
+          .with(env_filter);
+        set_global_default(subscriber).unwrap();
+      },
+      Environment::Production => {
+        let subscriber = builder.json().finish().with(env_filter);
+        set_global_default(subscriber).unwrap();
+      },
+    }
+  });
+}
+
+pub async fn create_app(listener: TcpListener, config: Config) -> Result<(), Error> {
+  // Postgres
+  info!("Preparing to run database migrations...");
+  let pg_pool = get_connection_pool(&config.db_settings).await?;
+
+  // Redis
+  let redis_client = redis::Client::open(config.redis_url)
+    .expect("failed to create redis client")
+    .get_connection_manager()
+    .await
+    .expect("failed to get redis connection manager");
+
+  let s3_client = get_aws_s3_client(&config.s3_setting).await?;
+
+  let state = AppState {
+    redis_client,
+    pg_pool,
+    s3_client,
+  };
+
+  let local_set = LocalSet::new();
+  let email_notifier = EmailNotifier;
+  let import_worker_fut = local_set.run_until(run_import_worker(
+    state.pg_pool.clone(),
+    state.redis_client.clone(),
+    Arc::new(state.s3_client.clone()),
+    Arc::new(email_notifier),
+    "import_task_stream",
+    30,
+  ));
+
+  let app = Router::new()
+    .route("/health", get(health_check))
```

Comment:
I dont think health check is needed, since it's a background task consumer and should not any user http. Then we can remove the whole axum dependency/setup

Reply:
Do we need this endpoint to check the worker server is up or not?

Reply:
Consider that this worker is independent and no other services needs to know whether the worker server is up or not, i would say that the healthcheck endpoint has very limited use.

Reply:
If we really need to monitor whether the worker is performing well or not exposing prometheus metric is probably a better way then having a health check endpoint.